Build warnings from LLVM

After #714 was merged, the code added in #715 compiles with warnings from LLVM.

Example:
```console
src/experimental/__p0009_bits/config.hpp:45:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
```

--- a/src/compton/config.h.in
+++ b/src/compton/config.h.in
@@ -6,8 +6,8 @@
  *         All rights reserved. */
 /*---------------------------------------------------------------------------*/
 
-#ifndef __compton_config_h__
-#define __compton_config_h__
+#ifndef rtt_compton_config_h
+#define rtt_compton_config_h
 
 #cmakedefine COMPTON_FOUND
 
@@ -22,7 +22,7 @@
 
 #endif
 
-#endif /* __compton_config_h__ */
+#endif /* rtt_compton_config_h */
 
 /*---------------------------------------------------------------------------*/
 /* end of compton/config.h */

--- a/src/diagnostics/config.h.in
+++ b/src/diagnostics/config.h.in
@@ -8,8 +8,8 @@
  *         All rights reserved. */
 /*---------------------------------------------------------------------------*/
 
-#ifndef __diagnostics_config_h__
-#define __diagnostics_config_h__
+#ifndef rtt_diagnostics_config_h
+#define rtt_diagnostics_config_h
 
 /* DRACO TIMING LEVEL */
 #cmakedefine DRACO_TIMING @DRACO_TIMING@
@@ -17,7 +17,7 @@
 /* USE PROCMON - snapshot in time memory reporting via getrusage */
 #cmakedefine USE_PROCMON @USE_PROCMON@
 
-#endif /* __diagnostics_config_h__ */
+#endif /* rtt_diagnostics_config_h */
 
 /*---------------------------------------------------------------------------*/
 /* end of diagnostics/config.h */

--- a/src/experimental/mdspan
+++ b/src/experimental/mdspan
@@ -53,6 +53,7 @@
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wunused-parameter"
 #pragma clang diagnostic ignored "-Wgnu-zero-variadic-macro-arguments"
+#pragma clang diagnostic ignored "-Wreserved-id-macro"
 #endif
 
 #include "__p0009_bits/accessor_basic.hpp"

